R/checks.R

```diff
@@ -0,0 +1,86 @@
+#Global variable specifying what metadata columns are absolutely required
+required_cols = c("sample_id","patient_id","pathology","seq_type","genome_build","pairing_status","Tumor_Sample_Barcode")
+
+#' Check GAMBL or other metadata for compatability with various features
+#'
+#' @param metadata_df Data frame output by get_gambl_metadata or some other source of metadata you plan to use
+#' @param to_check Specify one of "uniqueness", "colours" or "completeness" or leave empty to check all
+#' @param fix After identifying an issue, rerun this function with fix=TRUE to address errors (when possible). Currently this doesn't do anything. That's how I roll
+#' @param show_details Set to TRUE if you want the gory details about issues that are identified
+#'
+#' @return
+#' @export
+#' @import dplyr
+#'
+#' @examples
+check_gambl_metadata = function(metadata_df,to_check="all",show_details=FALSE,fix=FALSE){
+  
+  if(to_check == "all"){
+    to_check = c("uniqueness","colours","completeness")
+  }
+  if("completeness" %in% to_check){
+    if(any(!required_cols %in% colnames(metadata_df))){
+      missing = required_cols[!required_cols %in% colnames(metadata_df)]
+      message("FAIL! MISSING metadata columns:")
+      print(paste(missing,sep=" "))
+      return()
```

Comment:
What about adding here `else` part that will just say something alongside "Success! All required columns are present in the metadata". This way we are letting user know that the completeness check was performed and all is well